### ISSUE TITLE
Sharp fails to build on Codeship

### ISSUE THREAD

**Issue by AoDev** [lovell/sharp#222]
Hello, 

Has anyone had this problem? I have no problem running sharp locally or, we also use the Heroku Sharp buildpack. But, on codeship it fails to build. I tried with version 0.9.3 and 0.10.0.

This is the build log:

npm WARN package.json ***@1.0.0 No description
npm WARN package.json ***@1.0.0 No repository field.
npm WARN package.json ***@1.0.0 No README data
npm http GET https://registry.npmjs.org/sharp
npm http 200 https://registry.npmjs.org/sharp
npm http GET https://registry.npmjs.org/sharp/-/sharp-0.10.0.tgz
npm http 200 https://registry.npmjs.org/sharp/-/sharp-0.10.0.tgz
npm http GET https://registry.npmjs.org/bluebird
npm http GET https://registry.npmjs.org/color
npm http GET https://registry.npmjs.org/nan
npm http GET https://registry.npmjs.org/semver
npm http 200 https://registry.npmjs.org/color
npm http GET https://registry.npmjs.org/color/-/color-0.8.0.tgz
npm http 200 https://registry.npmjs.org/semver
npm http 200 https://registry.npmjs.org/nan
npm http GET https://registry.npmjs.org/semver/-/semver-4.3.4.tgz
npm http GET https://registry.npmjs.org/nan/-/nan-1.8.4.tgz
npm http 200 https://registry.npmjs.org/color/-/color-0.8.0.tgz
npm http 200 https://registry.npmjs.org/bluebird
npm http GET https://registry.npmjs.org/bluebird/-/bluebird-2.9.25.tgz
npm http 200 https://registry.npmjs.org/semver/-/semver-4.3.4.tgz
npm http 200 https://registry.npmjs.org/nan/-/nan-1.8.4.tgz
npm http 200 https://registry.npmjs.org/bluebird/-/bluebird-2.9.25.tgz
npm http GET https://registry.npmjs.org/color-convert
npm http GET https://registry.npmjs.org/color-string
npm http 200 https://registry.npmjs.org/color-convert
npm http 200 https://registry.npmjs.org/color-string
npm http GET https://registry.npmjs.org/color-convert/-/color-convert-0.5.2.tgz
npm http GET https://registry.npmjs.org/color-string/-/color-string-0.3.0.tgz
npm http 200 https://registry.npmjs.org/color-convert/-/color-convert-0.5.2.tgz
npm http 200 https://registry.npmjs.org/color-string/-/color-string-0.3.0.tgz
npm http GET https://registry.npmjs.org/color-name
npm http 200 https://registry.npmjs.org/color-name
npm http GET https://registry.npmjs.org/color-name/-/color-name-1.0.0.tgz
npm http 200 https://registry.npmjs.org/color-name/-/color-name-1.0.0.tgz

> sharp@0.10.0 install /home/rof/src/github.com/***/***/node_modules/sharp
> node-gyp rebuild

gyp http GET http://nodejs.org/dist/v0.10.25/node-v0.10.25.tar.gz
gyp http 200 http://nodejs.org/dist/v0.10.25/node-v0.10.25.tar.gz
make: Entering directory `/home/rof/src/github.com/***/***/node_modules/sharp/build'
CXX(target) Release/obj.target/sharp/src/common.o
../src/common.cc: In function sharp::ImageType sharp::DetermineImageType(void*, size_t):
../src/common.cc:40:68: error: vips_foreign_find_load_buffer was not declared in this scope
char const *load = vips_foreign_find_load_buffer(buffer, length);
^
../src/common.cc: In function VipsImage* sharp::InitImage(void*, size_t, VipsAccess):
../src/common.cc:62:83: error: vips_image_new_from_buffer was not declared in this scope
return vips_image_new_from_buffer(buffer, length, NULL, "access", access, NULL);
^
../src/common.cc: In function VipsImage* sharp::InitImage(const char*, VipsAccess):
../src/common.cc:94:65: error: too many arguments to function VipsImage* vips_image_new_from_file(const char*)
return vips_image_new_from_file(file, "access", access, NULL);
^
In file included from /usr/include/vips/vips.h:112:0,
from ../src/common.cc:3:
/usr/include/vips/image.h:420:12: note: declared here
VipsImage *vips_image_new_from_file( const char *filename );
^
../src/common.cc: In function VipsImage* sharp::InitImage(void*, size_t, VipsAccess):
../src/common.cc:63:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
make: *** [Release/obj.target/sharp/src/common.o] Error 1
make: Leaving directory `/home/rof/src/github.com/***/***/node_modules/sharp/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/rof/.nvm/v0.10.25/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:797:12)
gyp ERR! System Linux 3.13.0-35-generic
gyp ERR! command "node" "/home/rof/.nvm/v0.10.25/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/rof/src/github.com/***/***/node_modules/sharp
gyp ERR! node -v v0.10.25
gyp ERR! node-gyp -v v0.12.2
gyp ERR! not ok 
npm ERR! sharp@0.10.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sharp@0.10.0 install script.
npm ERR! This is most likely a problem with the sharp package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls sharp
npm ERR! There is likely additional logging output above.

npm ERR! System Linux 3.13.0-35-generic
npm ERR! command "/home/rof/.nvm/v0.10.25/bin/node" "/home/rof/.nvm/v0.10.25/bin/npm" "install"
npm ERR! cwd /home/rof/src/github.com/***/***
npm ERR! node -v v0.10.25
npm ERR! npm -v 1.3.24
npm ERR! code ELIFECYCLE
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/rof/src/github.com/***/***/npm-debug.log
npm ERR! not ok code 0

**Comment by lovell** [comment]
Hello, it looks like a version of libvips prior to the required [7.40.0](https://github.com/lovell/sharp#prerequisites) is in use.

(I think Codeship uses Ubuntu 14.04, which by default provides [7.38.5](http://packages.ubuntu.com/trusty/libvips-dev) via `apt`.)

**Comment by AoDev** [comment]
I have commented this issue in a bug raised by someone a long time ago. 

I think it's in our interest to give some "up" there: :)

https://bugs.launchpad.net/ubuntu/+source/vips/+bug/1377030

**Comment by lovell** [comment]
If Codeship support PPAs (or at least have a list of "trusted" PPAs) then you may be able to use http://sharp.dimens.io/en/latest/install/#ubuntu-lts

**Comment by lovell** [comment]
@AoDev were you able to make any progress on this?

**Comment by AoDev** [comment]
@lovell  Short answer is no.

Since then I wrote to Codeship and this was their answer:

> Ahoy Kev, 
unfortunately the latest version of libvips on Ubuntu Trusty is too old for the library. See http://packages.ubuntu.com/search?keywords=libvips&searchon=names&suite=trusty&section=all for the details. 
We are working on fully customisable build VMs, but until then (or until Ubuntu releases an update) I can't install a newer version on the build VMs.
Best, 
Marko

In our project, we decided to take a different approach and we are not using sharp anymore. I don't know it customisable VM are available now.
